Update demographic info

roxelo opened the pull request:
- Updated the data for all the existing demographic graphs
- Added additional graphs
- Added some initial analysis and commentary on the results
- Added a way to have multiple data sets for the horizontal bar graph

roxelo committed:
Update data for demographic section and add missing graphs

roxelo committed:
Add horizontal bar graph that handles multi series

roxelo committed:
Update some graphs and add discussion and analysis

AkshayPall commented on content/background.html:
```diff
@@ -22,55 +21,71 @@ <h3> Gender </h3>
-  There was a significant number of students Caucasians, South Asians, and East
-  Asians.
+  There was a significant number of students Caucasians, and East
+  Asians, although it is interesting to note that most Women were East Asian.
```
nit: Women should be lowercase.

AkshayPall commented on content/background.html:
```diff
@@ -1,19 +1,18 @@
-  One student preferred not to disclose.
 </p>
 <div class="half second">
   <span class="graph-title">Gender proportion in class</span>
```
The entry for "non-binary" overlaps with "prefer not to disclose" on my display (13" 2015 mbp pro). 
<img width="330" alt="Screen Shot 2021-05-26 at 1 14 52 PM" src="https://user-images.githubusercontent.com/10256087/119703267-5d660800-be24-11eb-95f1-d55fe52162d8.png">

roxelo replied:
Yes, I am aware of this. I was planning to fix this in a later version, cause I think the pi graph still makes sense to use, but I guess a horizontal bar graph does the same job too

AkshayPall replied:
It's alright, we can mark a TODO for this. Pie does make the most sense

AkshayPall commented on content/background.html:
```diff
@@ -22,55 +21,71 @@ <h3> Gender </h3>
-  students.
+  The class saw a 2.1% decrease in international students from 1A (15%) to 4B.
+  This could be due to the higher tuition costs for international students, especially in Engineering.
+  International term tuition fee, prior to the academic year 2018/19 where it was subject to a 62.1% fee increase, 
```
nit: should be "fees"

AkshayPall commented on data/background.ts:
```diff
@@ -1,144 +1,391 @@
+  "name": "Heterosexual",
+  "value": 63
+}, {
+  "name": "Homosexual",
```
I fear that since there are only 2 entries each for Homosexual and  Bisexual, it might oust people who don't want to be? Can we anonymize this by either
1. Adding noise
2. Grouping both categories together into "non-hetero"?

either way, we then should disclose the strategy we used to protect people in the description

roxelo replied:
I don't see how it could out someone, cause it doesn't map LGTBQ+ with any other aspects. So we only know there are 2 homosexual students in our class but not who or any other aspects they share.

If we group the two categories together I would stick with naming it "LGBTQ+"

AkshayPall replied:
that's true. It might not oust someone because this data is in isolation from all the other graphs so there's no way to link these 2 students. Adding and disclosing noise might still make it more difficult for students to determine the orientation of people around them. Like you said, we can discuss this tomorrow

AkshayPall commented on main.ts:
```diff
@@ -218,12 +243,29 @@ function renderAcademics(options) {
+  renderHorizontalBarChat(d3.select('#mother-tongue'), MOTHER_TONGUE, 400, 300, true);
+  renderPieChart(d3.select('#programming'), PROGRAMMING, options.width * 0.75, options.width * 0.75);
+  renderPieChart(d3.select('#cat-or-dog'), CAT_OR_DOG, options.width * 0.75, options.width * 0.75);
+  renderDotPlot(d3.select('#admission-average'), ADMISSION_AVERAGE, options.width, 400, {
```
Would a histogram or bar graph suit this better?  We should leave a TODO to try out the histogram once #3 is merged.

roxelo replied:
It would. Will do that once #3 merges

AkshayPall commented on content/background.html:
```diff
@@ -101,13 +158,118 @@ <h3> Level of education </h3>
+  Among other reasons, this could also be due to transfer students who joined the program later.
+</p>
+<div class="half second">
+  <span class="graph-title">Students New Programming in High School</span>
```
nit: should say "Knew how to program"

roxelo replied:
Oops

AkshayPall reviewed (changes requested):
The anonymity of the sexuality graph is my only concern. There are other nits also that we can fix now or later. And thank you for adding the new charts!

roxelo committed:
fix nits

roxelo commented:
@AkshayPall I addressed most of your nits in the latest commit. Two of your comments I will fix in a later stage (pie chart issue and using histogram) as for the sexual orientation comment I don't think it is a problem, but we can also discuss this at tmr's meeting

AkshayPall reviewed (approved):
👍
